iceoryx_examples/automotive_soa/topic.hpp

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_TOPIC_HPP
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_TOPIC_HPP
+
+#include "iceoryx_hoofs/cxx/serialization.hpp"
+
+#include <chrono>
+#include <cstdint>
+
+constexpr uint32_t ONE_KILOBYTE = 1000U;
+
+struct AddRequest
+{
+    uint64_t addend1{0};
+    uint64_t addend2{0};
+};
+
+struct AddResponse
+{
+    uint64_t sum{0};
+};
+
+struct Topic
+{
+    uint32_t counter{0};
+};
+
+struct TimestampTopic1Byte
+{
+    // Printed to console
+    uint32_t counter{0};
+    std::chrono::time_point<std::chrono::steady_clock> sendTimestamp;
+
+    // Not printed to console
+    static constexpr uint32_t payloadSizeInBytes{1};
+    char data[payloadSizeInBytes];
+    uint32_t subPackets{0};
+
+    /// @todo #1332 Use cxx::Serialization and implement serialization for std::chrono::time_point?
```

Comment:
@elBoberido 

Just some small questions in this PR, no need for a full review:

1) I didn't use `cxx::Serialization` for the UDS `EventSender`, which uses a `std::string`-based API instead I did:

```
std::string tempBuffer;
auto sendTimeStampNs = userSamplePtr->sendTimestamp.time_since_epoch().count();
tempBuffer.append(reinterpret_cast<char*>(&sendTimeStampNs), 8);
```

Did you consider sending arbitrary types in your new serialization design?

Reply:
Ok, thanks, sounds great.

I know what you mean, it's definitely not the most elegant solution. However, as the `Event{Publisher,Subscriber}Uds` are just for performance comparison, I would prefer to keep the implementation as easy as possible. I don't think null termination is an issue because [it's possible to `append()` `0`s](https://godbolt.org/z/fjGrbhdo6). The main disadvantage is that `iox::posix::UnixDomainSocket` provides an API which is bases on `std::string`.

Reply:
@mossmaurice I did not look at the code, just your comment and did not have the context. Of course that works if you use it only to transfer the data and don't actually use the string as string but as byte array.